### PR TITLE
Work in progress

### PR DESCRIPTION
**Pull request by karolk**
# TODO

~~1. use and pass beacon GUID~~
~~2. create impression ID [NOT FOR NOW]~~
3. revisit shouldRedirect and shouldTrack logic
4. decide about creative ID
5. amp tests
~~6. remove GUID from go.redirecting links~~
7. use amp preconect
~~8. do we need xcust? NO~~
~~9. Should amp work in IE? NO! (acc to docs)~~
10. Can we track without affiliatising (falling back to NA)

### CODE REVIEW

File: extensions/amp-skimlinks/0.1/page.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  isAffiliatableUrl(url) {
+    return this.affiliateDomains.indexOf(Util.domain(url)) >= 0
+  }
+  
+  isNAUrl(url) {
+    return this.nonAffiliateDomains.indexOf(Util.domain(url)) >=0 
+  }
+  
+  isSameDomainUrl(url) {
+    return this.contextWin.location.hostname === Util.domain(url)
```
Comment (lepunk) [inline extensions/amp-skimlinks/0.1/page.js:86]: probs should deal with subdomains

File: extensions/amp-skimlinks/0.1/page.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        if (numId) {
+          slmId = numId
+        }
+      }
+    }
+    return slmId
+  }
+  
+  isAffiliatableUrl(url) {
+    return this.affiliateDomains.indexOf(Util.domain(url)) >= 0
```
Comment (lepunk) [inline extensions/amp-skimlinks/0.1/page.js]: this returns true to http://notexactlyebay.com

File: extensions/amp-skimlinks/0.1/skimlinks.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      //Url is affiliatable or unknown
+      return !this.page.isSameDomainUrl(url) && (this.page.isAffiliatableUrl(url) ||
+        (this.affiliateUnkownLinks && !this.page.isNAUrl(url)))
+    }
+    
+    this.rewriteLinkUrl = (link) => {
+      let origUrl = link.getAttribute('href')
+      let newUrl = this.createAffiliateUrl(origUrl)
+      link.href = newUrl
+      setTimeout(() => link.setAttribute("href", origUrl), 100)
```
Comment (lepunk) [inline extensions/amp-skimlinks/0.1/skimlinks.js:26]: not sure if amp deals with IE but if so for them we need longer timeout to swap back

File: extensions/amp-skimlinks/0.1/page.js (hunk 2 above)
Comment (karolk) [inline extensions/amp-skimlinks/0.1/page.js]: good catch

File: extensions/amp-skimlinks/0.1/page.js (hunk 1 above)
Comment (karolk) [inline extensions/amp-skimlinks/0.1/page.js:86]: There are 2 cases, lower level subdomain which is easy (yahoo.com and news.yahoo.com) and sibling subdomain which hard (news.about.com, sport.about.com). Which one did you mean?

Comment (lepunk) [inline extensions/amp-skimlinks/0.1/page.js:86]: ideally both :P

Comment (karolk) [inline extensions/amp-skimlinks/0.1/page.js:86]: Implemented and the easy one for now.

File: extensions/amp-skimlinks/0.1/skimlinks.js (hunk 3 above)
Comment (karolk) [inline extensions/amp-skimlinks/0.1/skimlinks.js:26]: Amp does not work with non-webkit/blink browsers for now.

File: extensions/amp-skimlinks/0.1/page.js (hunk 2 above)
Comment (karolk) [inline extensions/amp-skimlinks/0.1/page.js]: I don't think it does, this is basically `['a.com', 'b.com'].indexOf('c.com')`, so only a full string match will satisfy the condition.

**Pull request closed without merging**